src/components/tooltip/Tooltip.tsx

```diff
@@ -1,99 +1,127 @@
-import * as React from "react";
-import { useTooltipHover } from "./hooks";
-import { StyledTooltip, TooltipWrapper } from "./styles";
-
-export type Props = {
-  caption: React.ReactNode;
-  position?: "left" | "right";
-  bottom?: boolean;
-  status?: "danger" | "notification" | "success" | "warning";
-  children?: React.ReactNode;
-  showTimeout?: number;
-  hideTimeout?: number;
-  minWidth?: string;
-};
+import {
+  cloneElement,
+  ComponentProps,
+  forwardRef,
+  HTMLProps,
+  isValidElement,
+  ReactNode,
+  RefAttributes,
+} from "react";
+import {
+  useMergeRefs,
+  FloatingPortal,
+  useTransitionStyles,
+  FloatingArrow,
+} from "@floating-ui/react";
+import { When } from "react-if";
+import { isForwardRef } from "react-is";
+import { TooltipContext, useTooltipContext } from "./context";
+import { getArrowFillColor, StyledTooltip } from "./styles";
+import { TooltipOptions, useTooltip } from "./useTooltip";
 
 export function Tooltip({
-  status,
-  position,
-  caption,
-  bottom = false,
   children,
-  hideTimeout = 100,
-  showTimeout = 300,
-  minWidth = "0",
-}: Props) {
-  const tooltipRef = React.useRef<HTMLDivElement>(null);
-  const wrapperRef = React.useRef<HTMLDivElement>(null);
-
-  const [tooltipPosition, setPosition] = React.useState({
-    marginTop: 0,
-    width: 0,
-  });
+  ...options
+}: TooltipOptions & { children: ReactNode }) {
+  // This can accept any props as options, e.g. `placement`,
+  // or other positioning options.
+  const tooltip = useTooltip(options);
+  return (
+    <TooltipContext.Provider value={tooltip}>
+      {children}
+    </TooltipContext.Provider>
+  );
+}
 
-  const { isHovered, updateIsHovered } = useTooltipHover();
+export const TooltipTrigger = forwardRef<HTMLElement, HTMLProps<HTMLElement>>(
+  ({ children, ...props }, propertyRef) => {
+    const context = useTooltipContext();
+    // Casting, because React types are lacking. Children prop is typed as ReactNode which doesn't allow refs, which is not true
+    const childrenRef =
+      (children as RefAttributes<HTMLElement> | undefined)?.ref ?? null;
```

Comment:
What about using `@ts-expect-error` instead? This way we would know once the React types has been fixed and receive error then.

Reply:
done